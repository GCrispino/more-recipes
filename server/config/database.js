const path = require("path");
/**
 * The different database configurations for different environments.
 */
const config = {
  development: {
    username: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_DATABASE,
    host: process.env.DB_HOST,
    dialect: process.env.DB_DIALECT,
    logging: false,
    operatorAliases: false
  },
  test: {
    dialect: "sqlite",
    storage: "database.sqlite",
    logging: false,
    operatorAliases: false
  },
  production: {
    connection_uri: process.env.DATABASE_URL,
    logging: false,
    operatorAliases: false
  }
};

module.exports = config;
