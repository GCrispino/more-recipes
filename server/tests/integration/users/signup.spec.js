import supertest from "supertest";
import faker from "faker";
import app from "../../../index";
import { User } from "../../../database/models";

describe("The user signup test", () => {
  test("should register a new user", async () => {
    const fakeUser = {
      name: "Test user",
      email: faker.internet.email(),
      password: "1234567"
    };
    const response = await supertest(app)
      .post("/api/v1/users/signup")
      .send(fakeUser)
      .expect(200);

    expect(response.body.data.user.email).toBe(fakeUser.email);
    expect(response.body.data.user.name).toBe(fakeUser.name);
    expect(response.body.data.access_token).toBeDefined();

    const userfromDB = User.find({ where: { email: fakeUser.email } });

    expect(userfromDB).toBeTruthy();
  });

  test("Should return validation error for duplicate email", async () => {
    const fakeUser = {
      name: "Test user",
      email: faker.internet.email(),
      password: "12312323"
    };

    await User.create(fakeUser);

    const response = await supertest(app)
      .post("/api/v1/users/signup")
      .send(fakeUser)
      .expect(422);

    expect(response.body).toMatchSnapshot();
    expect(response.body.status).toBe("fail");
    expect(response.body.data).toEqual({
      errors: ["A user with this email already exists."]
    });
  });
});
