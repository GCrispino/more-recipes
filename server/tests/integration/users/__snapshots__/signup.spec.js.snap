// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The user signup test Should return validation error for duplicate email 1`] = `
Object {
  "data": Object {
    "errors": Array [
      "A user with this email already exists.",
    ],
  },
  "status": "fail",
}
`;
